lib/features/base/base_mailbox_controller.dart

```diff
@@ -554,4 +601,23 @@ abstract class BaseMailboxController extends BaseController {
     mailboxNode = personalMailboxTree.value.findNodeOnFirstLevel((node) => node.item.name?.name.toLowerCase() == name);
     return mailboxNode;
   }
+
+  void copySubaddressAction(BuildContext context, String subaddress) {
+    Clipboard.setData(ClipboardData(text: subaddress));
+    appToast.showToastSuccessMessage(context, AppLocalizations.of(context).emailSubaddressCopiedToClipboard);
+  }
+
+  String getSubaddress(String userEmail, String folderName) {
+    if (folderName.isEmpty) {
+      throw const FormatException('folder name should not be empty');
```

Comment:
Should define a separate exception for it. It could be: `FolderEmptyNameException`